pkg/container/types/types.go

```diff
@@ -104,7 +104,8 @@ type Type struct {
 	// todo: need to add new attribute DisplayWidth ?
 	Width int32
 	// Scale means number of fractional digits for decimal, timestamp, float, etc.
-	Scale int32
+	Scale   int32
+	NotNull bool
```

Comment:
Don't put it here. This 1 byte actually takes 4 bytes considering alignment.

Should replace `dummy1` or `dummy2`.